src/dom.js

```diff
@@ -137,32 +140,58 @@ export function elementByTag(element, tagName) {
  * @return {?Element}
  */
 export function childElement(parent, callback) {
-  const children = parent.children;
-  for (let i = 0; i < children.length; i++) {
-    if (callback(children[i])) {
-      return children[i];
+  for (let child = parent.firstElementChild; child;
+      child = child.nextElementSibling) {
+    if (callback(child)) {
+      return child;
     }
   }
   return null;
 }
 
+/**
+ * @type {boolean|undefined}
+ * @visiblefortesting
+ */
+let scopeSelectorSupported;
+
+/**
+ * @param {boolean|undefined} val
+ * @visiblefortesting
+ */
+export function setScopeSelectorSupportedForTesting(val) {
+  scopeSelectorSupported = val;
+}
+
+/**
+ * @return {boolean}
+ */
+function isScopeSelectorSupported() {
+  try {
+    document.querySelector(':scope');
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
 
 /**
- * Finds the first child element that has the specified attribute, optionally
- * with a value.
+ * Finds the first child element that has the specified attribute.
  * @param {!Element} parent
  * @param {string} attr
- * @param {string=} opt_value
  * @return {?Element}
  */
-export function childElementByAttr(parent, attr, opt_value) {
+export function childElementByAttr(parent, attr) {
+  if (scopeSelectorSupported == null) {
+    scopeSelectorSupported = isScopeSelectorSupported();
+  }
+  if (scopeSelectorSupported) {
+    return parent.querySelector(':scope > [' + attr + ']');
```

Comment:
if i remember correctly its because of "Selectors are evaluated against a given element in the context of the entire DOM tree in which the element is located."